src/common/Utilities.py

```diff
@@ -544,3 +544,200 @@ def is_operator_adjoint(operator, num_tests = 5, max_err = 10e-5, verbose = True
+        numpy.testing.assert_array_equal(tmp1.as_array(), image2.as_array())
+    
+    def test_add_datacontainer(self):
+        if hasattr(self, 'cwd'):
+            os.chdir(self.cwd)
+        image1 = self.image1
+        image2 = self.image2
+        image1.fill(0.)
+        image2.fill(1.)
+        tmp = image1 + image2
+
+        numpy.testing.assert_array_almost_equal(
+            numpy.ones(image1.shape, dtype=numpy.float32), tmp.as_array()
+            )
+    
+        tmp1 = image1.add(image2)
+        
+        numpy.testing.assert_array_almost_equal(
+            numpy.ones(image1.shape, dtype=numpy.float32), tmp1.as_array()
+            )
+        
+        tmp1.fill(2.)
+        image1.add(image2, out=tmp1)
+        
+        numpy.testing.assert_array_almost_equal(
+            numpy.ones(image1.shape, dtype=numpy.float32), tmp1.as_array()
+            )
+        
+    
+    def test_subtract_scalar(self):
+        if hasattr(self, 'cwd'):
+            os.chdir(self.cwd)
+        image1 = self.image1
+        image2 = self.image2
+        image1.fill(2)
+        image2.fill(1)
+        
+        tmp = image1 - 1.
+        numpy.testing.assert_array_equal(image2.as_array(), tmp.as_array())
+    
+        tmp1 = image1.subtract(1.)
+        numpy.testing.assert_array_equal(tmp.as_array(), tmp1.as_array())
+        
+        tmp1.fill(0)
+        image1.subtract(1., out=tmp1)
+        numpy.testing.assert_array_equal(tmp1.as_array(), image2.as_array())
+
+    def test_subtract_datacontainer(self):
+        if hasattr(self, 'cwd'):
+            os.chdir(self.cwd)
+        image1 = self.image1
+        image2 = self.image2
+        
+        tmp = image1 - image2
+
+        numpy.testing.assert_array_almost_equal(
+            numpy.zeros(image1.shape, dtype=numpy.float32), tmp.as_array()
+            )
+    
+        tmp1 = image1.subtract(image2)
+        
+        numpy.testing.assert_array_almost_equal(
+            numpy.zeros(image1.shape, dtype=numpy.float32), tmp1.as_array()
+            )
+        
+        tmp1.fill(2.)
+        image1.subtract(image2, out=tmp1)
+        
+        numpy.testing.assert_array_almost_equal(
+            numpy.zeros(image1.shape, dtype=numpy.float32), tmp1.as_array()
+            )
+
+    def test_division_by_scalar_zero(self):
+        if hasattr(self, 'cwd'):
+            os.chdir(self.cwd)
+        try:
+            self.image1 / 0.
+            self.assertFalse(True)
+        except ZeroDivisionError:
+            self.assertTrue(True)
+        except error:
+            self.assertTrue(True)
+    
+    def test_division_by_datacontainer_zero(self):
+        if hasattr(self, 'cwd'):
+            os.chdir(self.cwd)
+        try:
+            self.image2 *= 0
+            tmp = self.image1 / self.image2
+            self.assertFalse(True)
+        except ZeroDivisionError:
+            self.assertTrue(True)
+        except error:
+            self.assertTrue(True)
```

Comment:
This is testing for exceptions to be thrown, but see https://github.com/SyneRBI/SIRF/issues/863. want to merge this as-is and change it later?

Reply:
I will change it when we take a decision on how to handle division by zero, #864.
So far it is a nice test that checks if the exception is thrown when expected.